### PR TITLE
Add a mutable subscript to `Graph` when `V` is optional.

### PR DESCRIPTION
**Pull request by grynspan**
This PR modifies the subscript operator of the `Graph` pseudo-collection type such that, when `V` is optional, it is settable (with intermediate values defaulting to `nil`). This change simplifies some code that uses these subscripts.

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

File: Sources/Testing/Support/Graph.swift [hunk 1]
```diff
@@ -223,6 +223,55 @@ extension Graph {
     self.init(value: nil, children: children)
   }
 
+  /// Get or set the leaf value at the node identified by the specified sequence
+  /// of keys.
```
Comment (stmontgomery) [inline Sources/Testing/Support/Graph.swift:227]: I've seen this documented in [other](https://developer.apple.com/documentation/swift/dictionary/subscript(_:)-8rfql) mutable subscripts as "**Accesses** the value … for reading and writing", using the verb 'access'.

File: Sources/Testing/Support/Graph.swift [hunk 2]
```diff
@@ -236,9 +285,10 @@ extension Graph {
   /// intermediate to it, the graph is not modified. To add a value when none
   /// previously exists, use ``insertValue(_:at:intermediateValue:)``.
   ///
-  /// - Complexity: O(*m* + *n*) where *n* is the number of elements in
-  ///   `keyPath` and *m* is the number of children at the penultimate node in
-  ///   `keyPath`.
+  /// - Complexity: To get a value, O(*n*) where *n* is the number of elements
+  ///   in `keyPath`. To set a value, O(*m* + *n*) where *n* is the number of
+  ///   elements in `keyPath` and *m* is the number of children at the
+  ///   penultimate node in `keyPath`.
```
Comment (stmontgomery) [inline Sources/Testing/Support/Graph.swift:291]: Why include the "To **get** a value..." sentence in this `updateValue` method, if it's focused on setting a value?

Comment (grynspan) [reply Sources/Testing/Support/Graph.swift:291]: This was a pasting error. :)

**Comment by grynspan** [comment]
@swift-ci please test

**Review by stmontgomery — APPROVED** [review]

File: Sources/Testing/Support/Graph.swift (hunk 1 above)
Comment (grynspan) [reply Sources/Testing/Support/Graph.swift:227]: The grammar gets a bit hard to parse if I change it because "for reading and writing" ends up quite far from "Accesses" so I'm just gonna leave it as-is.
